lib/medium-admin.php

```diff
@@ -136,10 +136,16 @@ public static function add_meta_boxes_post($post) {
   public static function save_post($post_id, $post) {
     if (defined('DOING_AUTOSAVE') && DOING_AUTOSAVE) return;
 
+    $allowed_post_types = (array) apply_filters('medium_allowed_post_types', array('post'));
+
+    if (!in_array(get_post_type($post_id), $allowed_post_types)) {
+      return;
+    }
+
     $medium_post = Medium_Post::get_by_wp_id($post_id);
 
     // If this post has already been sent to Medium, no need to do anything.
-    if ($medium_post->id) return;
+    if (!empty($medium_post->id)) return;
```

Comment:
If id hasn't been set, it should be null, which is falsey. I'd really like to see what your wp_postmeta value for the `medium_post` key looks like, that this would solve the problem, because:

```php
class Test {
    public $id;
    public function __construct($id) {
        $this->id = $id;
    }
}

$x = new Test("12");
if ($x->id) {
    print_r("HELLO");
}

$x = new Test("");
if ($x->id) {
    print_r("WORLD");
}

$y = unserialize(serialize($x));
if ($x->id) {
    print_r("AFTER");
```

only prints 'Hello'